File: src/main/java/org/example/toworkspring/Controllers/TrackController.java
package org.example.toworkspring.Controllers;

import org.example.toworkspring.DB;
import org.example.toworkspring.Utils;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.sql.*;
import java.util.List;
import java.util.Map;

@RestController
public class TrackController {
    @GetMapping("/")
    public List<Map<String, Object>> tracks() {
        ResultSet resultSet = null;
        Statement statement = null;
        Connection connection = null;

        try {
            connection = DB.connect();
            statement = connection.createStatement();
            resultSet = statement.executeQuery("select * from appuser");

            List<Map<String, Object>> json = Utils.serializeResultSet(resultSet);

            return json;
        } catch (SQLException e) {
            System.err.println(e.getMessage());
        } finally {
            try { if (resultSet != null) resultSet.close(); } catch (Exception e) {System.err.println(e.getMessage());}
            try { if (statement != null) statement.close(); } catch (Exception e) {System.err.println(e.getMessage());}
            try { if (connection != null) connection.close(); } catch (Exception e) {System.err.println(e.getMessage());}
        }
        return null;
    }
}

File: src/main/kotlin/org/example/toworkspring/controllers/TrackController.kt
package org.example.toworkspring.controllers

import org.example.toworkspring.models.Module
import org.example.toworkspring.models.Page
import org.example.toworkspring.models.Track
import org.example.toworkspring.services.TrackService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import javax.validation.constraints.Max
import javax.validation.constraints.Min

@RestController
class TrackController(private val trackService: TrackService) {
    @GetMapping("/getTracks")
    fun getTracks(
        @RequestParam(value = "page", defaultValue = "0") @Min(0) page: Int,
        @RequestParam(value = "pageSize", defaultValue = "20") @Min(0) @Max(50) pageSize: Int
    ): List<Track> {
        return trackService.getAllTracks(page, pageSize).toList()
    }

    @GetMapping("/getModules")
    fun getModulesInTrack(
        @RequestParam(value = "idTrack") idTrack: Int
    ): List<Module> {
        return trackService.getModulesInTrack(idTrack)
    }

    @GetMapping("/getPages")
    fun getPagesInModules(
        @RequestParam(value = "numberModule", defaultValue = "1") numberModule: Int,
        @RequestParam(value = "idTrack", defaultValue = "1") idTrack: Int
    ): List<Page> {
        return trackService.getPagesInModules(numberModule.toLong(), idTrack.toLong())
    }

    @GetMapping("/updateProgress")
    fun updateProgress(
        @RequestParam(value = "idUser") idUser: Int,
        @RequestParam(value = "idTrack") idTrack: Int,
        @RequestParam(value = "numberModule") numberModule: Int,
        @RequestParam(value = "numberPage") numberPage: Int,
    ) {
        trackService.updateProgress(idUser, idTrack, numberModule, numberPage)
    }
}
